pydantic/schema.py

```diff
@@ -0,0 +1,392 @@
+    model_name_map: Dict[Type['main.BaseModel'], str],
+    schema_overrides=False,
+    ref_prefix='#/definitions/',
+) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    definitions = {}
+    if len(sub_fields) == 1:
+        return field_type_schema(
+            sub_fields[0],
+            by_alias=by_alias,
+            model_name_map=model_name_map,
+            schema_overrides=schema_overrides,
+            ref_prefix=ref_prefix,
+        )
+    else:
+        sub_field_schemas = []
+        for sf in sub_fields:
+            sub_schema, sub_definitions = field_type_schema(
+                sf,
+                by_alias=by_alias,
+                model_name_map=model_name_map,
+                schema_overrides=schema_overrides,
+                ref_prefix=ref_prefix,
+            )
+            definitions.update(sub_definitions)
+            sub_field_schemas.append(sub_schema)
+        return {'anyOf': sub_field_schemas}, definitions
+
+
+def field_singleton_schema(  # noqa: C901 (ignore complexity)
+    field: Field,
+    *,
+    by_alias: bool,
+    model_name_map: Dict[Type['main.BaseModel'], str],
+    schema_overrides=False,
+    ref_prefix='#/definitions/',
+) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    definitions = {}
+    if field.sub_fields:
+        return field_singleton_sub_fields_schema(
+            field.sub_fields,
+            by_alias=by_alias,
+            model_name_map=model_name_map,
+            schema_overrides=schema_overrides,
+            ref_prefix=ref_prefix,
+        )
+    if field.type_ is Any:
+        return {}, definitions  # no restrictions
+    f_schema = {}
+    if issubclass(field.type_, Enum):
+        f_schema.update({'enum': [item.value for item in field.type_]})
+        # Don't return immediately, to allow adding specific types
+    # For constrained strings
+    if hasattr(field.type_, 'min_length'):
+        if field.type_.min_length is not None:
+            f_schema.update({'minLength': field.type_.min_length})
+    if hasattr(field.type_, 'max_length'):
+        if field.type_.max_length is not None:
+            f_schema.update({'maxLength': field.type_.max_length})
+    if hasattr(field.type_, 'regex'):
+        if field.type_.regex is not None:
+            f_schema.update({'pattern': field.type_.regex.pattern})
+    # For constrained numbers
+    if hasattr(field.type_, 'gt'):
+        if field.type_.gt is not None:
+            f_schema.update({'exclusiveMinimum': field.type_.gt})
+    if hasattr(field.type_, 'lt'):
+        if field.type_.lt is not None:
+            f_schema.update({'exclusiveMaximum': field.type_.lt})
+    if hasattr(field.type_, 'ge'):
+        if field.type_.ge is not None:
+            f_schema.update({'minimum': field.type_.ge})
+    if hasattr(field.type_, 'le'):
+        if field.type_.le is not None:
+            f_schema.update({'maximum': field.type_.le})
+    # Sub-classes of str must go before str
+    if issubclass(field.type_, EmailStr):
```

Comment:
again some kind of iterable and loop would be much better here.

Reply:
Done